provider-blackduck/src/main/java/com/synopsys/integration/alert/provider/blackduck/processor/message/service/BlackDuckMessageBomComponentDetailsCreator.java

```diff
@@ -148,10 +154,20 @@ private ComponentVulnerabilities retrieveComponentVulnerabilities(ProjectVersion
         }
         return blackDuckApiClient.getAllResponses(bomComponent.metaPolicyRulesLink())
                    .stream()
-                   .map(policyDetailsCreator::toComponentPolicy)
+                   .map(componentPolicyRulesView -> policyDetailsCreator.toComponentPolicy(componentPolicyRulesView, getPolicyRuleView(componentPolicyRulesView.getName())))
                    .collect(Collectors.toList());
     }
 
+    private Optional<PolicyRuleView> getPolicyRuleView(String policyName) {
+        Optional<PolicyRuleView> policyRuleView;
+        try {
+            policyRuleView = policyRuleService.getPolicyRuleViewByName(policyName);
+        } catch (IntegrationException e) {
+            return Optional.empty();
+        }
+        return policyRuleView;
+    }
```

Comment:
This logic seems like it would fit better in `BlackDuckComponentPolicyDetailsCreator`. 

Also, the Alert team prefers `fetch` or `retrieve` instead of `get` as a prefix for this type of method.